msim-tokio/src/sim/net.rs

```diff
@@ -112,7 +112,7 @@ impl TcpListener {
     }
 
     pub fn from_std(listener: std::net::TcpListener) -> io::Result<TcpListener> {
-        let fd: OwnedFd = listener.as_raw_fd().into();
+        let fd: OwnedFd = listener.into_raw_fd().into();
```

Comment:
in the old code, the `listener` passed in stay alive and ran its drop handler, closing the fd.